tests/integration/models/test_restart.py

```diff
@@ -0,0 +1,151 @@
+    fluid = pp.FluidConstants(fluid_vals)
+
+    params = {
+        "suppress_export": False,  # Suppress output for tests
+        "material_constants": {"solid": solid, "fluid": fluid},
+        "uy_north": uy_north,
+        "max_iterations": 20,
+        "time_manager": pp.TimeManager(schedule=[0, 1], dt_init=0.5, constant_dt=True),
+        "restart_options": {
+            "restart": restart,
+            "pvd_file": current_dir + "/restart_reference/previous_data.pvd",
+            "times_file": current_dir + "/restart_reference/previous_times.json",
+        },
+    }
+    setup = TailoredPoromechanics(params)
+    return setup
+
+
+@pytest.mark.parametrize(
+    "solid_vals,north_displacement",
+    [
+        ({"porosity": 0.5}, 0.1),
+    ],
+)
+def test_restart_2d_single_fracture(solid_vals, north_displacement):
+    """Restart version of .test_poromechanics.test_2d_single_fracture.
+
+    Provided the exported data from a previous time step, restart the
+    simulaton, continue running and compare the final state and exported
+    vtu/pvd files with reference files.
+
+    This test also serves as minimal documentation of how to restart a model
+    in a practical situation.
+
+    Parameters:
+        solid_vals (dict): Dictionary with keys as those in :class:`pp.SolidConstants`
+            and corresponding values.
+        north_displacement (float): Value of displacement on the north boundary.
+        expected_x_y (tuple): Expected values of the displacement in the x and y.
+            directions. The values are used to infer sign of displacement solution.
+
+    """
+    # Setup and run model for full time interval.
+    setup = create_fractured_setup(solid_vals, {}, north_displacement, restart=False)
+    pp.run_time_dependent_model(setup, {})
+
+    # The run generates data for initial and the first two time steps.
+    # In order to use the data as restart and reference data, move it
+    # to a reference folder.
+    pvd_files = list(Path("./visualization").glob("*.pvd"))
+    vtu_files = list(Path("./visualization").glob("*.vtu"))
+    json_files = list(Path("./visualization").glob("*.json"))
+    for f in pvd_files + vtu_files + json_files:
+        dst = Path(current_dir) / Path("restart_reference") / Path(f.stem + f.suffix)
+        shutil.move(str(f), str(dst))
+
+    # Now use the reference data to restart the simulation.
+    setup = create_fractured_setup(solid_vals, {}, north_displacement, restart=True)
+    pp.run_time_dependent_model(setup, {})
+
+    # To verify the restart capabilities, test whether:
+    # - the states have been correctly initialized at restart time;
+    # - the follow-up time step has been computed correctely;
+    # - the overall solution stored in a global pvd file is compiled correctly;
+    # - the logging of times and step sizes is correct.
+
+    for ending in ["000001", "000002"]:
+        for i in ["1", "2"]:
+            assert _compare_vtu_files(
```

Comment:
What will a failure tell us? When will we need to recompute the reference files?